Fix two security vulnerabilities in irssi.

More information at:

  https://irssi.org/security/irssi_sa_2017_06.txt

Upstream patch at:

  https://github.com/irssi/irssi/commit/fb08fc7f1aa6b2e616413d003bf021612301ad55

--- irssi-1.0.2/src/irc/dcc/dcc-get.c.orig	2017-06-06 15:31:20.375677210 +0000
+++ irssi-1.0.2/src/irc/dcc/dcc-get.c	2017-06-06 15:33:51.035145241 +0000
@@ -382,6 +382,8 @@
 	if (*params[0] == '"') {
 		/* quoted file name? */
 		for (pos = 0; pos < paramcount-3; pos++) {
+			if (strlen(params[pos]) == 0)
+				continue;
 			if (params[pos][strlen(params[pos])-1] == '"' &&
 			    get_params_match(params, pos+1))
 				return pos+1;
@@ -428,6 +430,10 @@
 	int p_id = -1;
 	int passive = FALSE;
 
+	if (addr == NULL) {
+		addr = "";
+	}
+
 	/* SEND <file name> <address> <port> <size> [...] */
 	/* SEND <file name> <address> 0 <size> <id> (DCC SEND passive protocol) */
 	params = g_strsplit(data, " ", -1);
--- irssi-1.0.2/src/irc/dcc/dcc-resume.c.orig	2017-06-06 15:34:13.665661067 +0000
+++ irssi-1.0.2/src/irc/dcc/dcc-resume.c	2017-06-06 15:34:51.541188077 +0000
@@ -62,6 +62,8 @@
 	if (*params[0] == '"') {
 		/* quoted file name? */
 		for (pos = 0; pos < paramcount-2; pos++) {
+			if (strlen(params[pos]) == 0)
+				continue;
 			if (params[pos][strlen(params[pos])-1] == '"' &&
 			    get_params_match_resume(params, pos+1))
 				return pos+1;
